core/services/ocr2/delegate.go

```diff
@@ -111,8 +112,68 @@ func (d *Delegate) BeforeJobCreated(spec job.Job) {
 	// This is only called first time the job is created
 	d.isNewlyCreatedJob = true
 }
-func (d *Delegate) AfterJobCreated(spec job.Job)  {}
-func (d *Delegate) BeforeJobDeleted(spec job.Job) {}
+func (d *Delegate) AfterJobCreated(spec job.Job) {}
+func (d *Delegate) BeforeJobDeleted(jb job.Job) {
+	spec := jb.OCR2OracleSpec
+	if spec == nil {
+		d.lggr.Errorf("offchainreporting2.Delegate.BeforeJobDeleted called with wrong job type, ignoring non-OCR2 spec %v", jb)
+		return
+	}
+	if spec.Relay != relay.EVM {
+		return
+	}
+
+	chainID, err := spec.RelayConfig.EVMChainID()
+	if err != nil {
+		d.lggr.Errorf("OCR2 jobs spec missing chainID")
+		return
+	}
+	chain, err := d.chainSet.Get(big.NewInt(chainID))
+	if err != nil {
+		d.lggr.Error(err)
+		return
+	}
+	lp := chain.LogPoller()
+
+	var filters []string
+	switch spec.PluginType {
+	case job.OCR2VRF:
+		filters, err = ocr2vrfcoordinator.FilterNamesFromSpec(spec)
+		if err != nil {
+			d.lggr.Errorw("failed to derive ocr2vrf filter names from spec", "err", err, "spec", spec)
+		}
+	case job.OCR2Keeper:
+		filters, err = ocr2keeper.FilterNamesFromSpec(spec)
+		if err != nil {
+			d.lggr.Errorw("failed to derive ocr2keeper filter names from spec", "err", err, "spec", spec)
+		}
+	default:
```

Comment:
Doesn't the linter usually complain when this is incomplete?

Reply:
Ah, I see... looks like these are defined as separate consts for some reason, rather than an enum.  Probably should be an enum!

Reply:
Oh, nevermind...  they are an enum to protobuf and graphql, just can't be in golang since there are no native enums.   I think it can't be a type since the values have to match between go and these transport protocols.

Reply:
By 'enum' in Go, I mean the const iota pattern. It may recognize other cases too :shrug:.